### PR TITLE
[folly related] update to 06.03

### PR DESCRIPTION
**Pull request by c8ef**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/fizz/OpenSSLKeyExchange.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/bin/ld: /data/sandcastle/temp/skycastle/fbcode_builder_getdeps/installed/fizz/lib/libfizz.a(OpenSSLFactory.cpp.o): in function `std::_MakeUniq<fizz::openssl::OpenSSLECKeyExchange>::__single_object std::make_unique<fizz::openssl::OpenSSLECKeyExchange, int const&, int const&>(int const&, int const&) [clone .isra.0]':
+OpenSSLFactory.cpp:(.text+0x1b7): undefined reference to `vtable for fizz::openssl::OpenSSLECKeyExchange'
+/bin/ld: /data/sandcastle/temp/skycastle/fbcode_builder_getdeps/installed/fizz/lib/libfizz.a(Utils.cpp.o): in function `std::_MakeUniq<fizz::openssl::OpenSSLECKeyExchange>::__single_object std::make_unique<fizz::openssl::OpenSSLECKeyExchange, int const&, int const&>(int const&, int const&) [clone .isra.0]':
+Utils.cpp:(.text+0x18): undefined reference to `vtable for fizz::openssl::OpenSSLECKeyExchange'
+collect2: error: ld returned 1 exit status
+[1064/1080] Building CXX object CMakeFiles/fsdb_handler.dir/fboss/fsdb/server/ServiceHandler.cpp.o
+FAILED: CMakeFiles/fsdb_handler.dir/fboss/fsdb/server/ServiceHandler.cpp.o
```
Comment (MonicaLiu0311) [inline ports/fizz/OpenSSLKeyExchange.patch:22]: Is this why you applied this patch?

Comment (MonicaLiu0311) [reply ports/fizz/OpenSSLKeyExchange.patch:22]: We generally recommend deleting the content before `diff` to reduce the size of patch.

Comment (dg0yt) [reply ports/fizz/OpenSSLKeyExchange.patch:22]: The actual change is one line. Can't you add a link for explanation?

**Comment by MonicaLiu0311** [comment]
Except `libaio,liburing`, other features are tested successfully in the following triplet:
```
x64-windows
x64-windows-static
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**